unixfs/archive/tar/writer.go

```diff
@@ -19,18 +20,29 @@ import (
 // unixfs merkledag nodes as a tar archive format.
 // It wraps any io.Writer.
 type Writer struct {
-	Dag  mdag.DAGService
-	TarW *tar.Writer
+	Dag            mdag.DAGService
+	TarW           *tar.Writer
+	progressBarOut io.Writer
+	progressBar    *progressbar.ProgressBar
 
 	ctx cxt.Context
 }
 
 // NewWriter wraps given io.Writer.
-func NewWriter(ctx cxt.Context, dag mdag.DAGService, archive bool, compression int, w io.Writer) (*Writer, error) {
+func NewWriter(ctx cxt.Context, dag mdag.DAGService, archive bool, compression int, w io.Writer, progressBarOut io.Writer, size uint64) (*Writer, error) {
```

Comment:
and their maintenance. I know the codebase is not great in this regard, but we should always strive to make it better.